com.ca.lsp.cobol/lsp-service-cobol/src/main/java/com/ca/lsp/cobol/service/CopybookServiceImpl.java

```diff
@@ -135,120 +115,47 @@ private Path retrievePathOrNull(String filename, List<Path> datasetPathList) {
   }
 
   private List<Path> generatePathListFromSettings(String profile, List<String> datasetList) {
-    // can happen here that copybooks or internal structure is null
     return datasetList.stream()
         .map(
             it ->
                 Paths.get(
-                    getCopybookFolder(getWorkspaceFoldersAsPathList().get(0))
+                    getCopybookBaseFolder(getWorkspaceFoldersAsPathList().get(0))
                         + filesystemSeparator()
                         + profile
                         + filesystemSeparator()
                         + it))
-        .filter(Objects::nonNull)
         .collect(Collectors.toList());
   }
 
-  /**
-   * @param fileName copybook name
-   * @param targetFolderPath physical path of workspace where to search for the copybook
-   * @return Path of the found copybook in the target folder.
-   */
-  private Path applySearch(String fileName, Path targetFolderPath) {
-    try (Stream<Path> pathStream =
-        Files.find(
-            targetFolderPath,
-            100,
-            (path, basicFileAttributes) -> isValidFileFound(path.toFile(), fileName),
-            FileVisitOption.FOLLOW_LINKS)) {
-      return pathStream.findAny().orElse(null);
-    } catch (IOException e) {
-      log.error(e.getMessage());
-      return null;
-    }
-  }
-
   private List<Path> getWorkspaceFoldersAsPathList() {
     return Optional.ofNullable(getWorkspaceFolders())
         .map(Collection::stream)
         .orElseGet(Stream::empty)
         .filter(Objects::nonNull)
-        .map(this::resolveUriPath)
+        .map(this::resolveURI)
         .collect(Collectors.toList());
   }
 
   private List<WorkspaceFolder> getWorkspaceFolders() {
     return workspaceFolders;
   }
 
-  /**
-   * Normalize the URI defined in the workspace to get a NIO Path object that will be used within
-   * the FileSystemService, example: [input:
-   * file:///C:/Users/test/AppData/Local/Temp/WORKSPACE/COPYTEST.cpy] --> [output:
-   * C:/Users/test/AppData/Local/Temp/WORKSPACE/COPYTEST.cpy]
-   *
-   * @param it workspace folder
-   * @return the Path of the workspace folder
-   * @throws IllegalArgumentException if the URI of WorkspaceFolder is not valid
-   */
-  private Path resolveUriPath(WorkspaceFolder it) {
-    try {
-      return Paths.get(new URI(it.getUri()).normalize());
-    } catch (URISyntaxException e) {
-      log.error(e.getMessage());
-    }
-    return null;
+  private Path resolveURI(WorkspaceFolder workspaceFolder) {
+    return getPathFromURI(workspaceFolder.getUri());
   }
 
-  private Path getCopybookFolder(Path workspaceFolderPath) {
+  private Path getCopybookBaseFolder(Path workspaceFolderPath) {
     return Paths.get(workspaceFolderPath + filesystemSeparator() + COPYBOOK_FOLDER_NAME);
```

Comment:
`Paths#get()` documentation recommends to use `workspaceFolderPath.resolve(COPYBOOK_FOLDER_NAME)` here

Reply:
For what I read from the [documentation](https://docs.oracle.com/javase/7/docs/api/java/nio/file/Paths.html):

> Note that while this method is very convenient, using it will imply an assumed reference to the default FileSystem and limit the utility of the calling code. Hence it should not be used in library code intended for flexible reuse. A more flexible alternative is to use an existing Path instance as an anchor, such as:
>
>     Path dir = ...
>     Path path = dir.resolve("file");

the recommendation is to use ```dir.resolve``` to avoid problem with different filesystems, but cause in my implementation I have the method ```filesystemSeparator()``` that should avoid problems with different FS I think we can choose to modify or leave like this. BTW I will give a try as low priority request of change if you agree.